### DISCUSSION TITLE
Operations with crystal directions

### DISCUSSION THREAD

**Discussion by eugenianietov** [mtex-toolbox/mtex#597] [Ask Anything] [answered]
Hello all!

I have been working on operations with crystal directions following the example included in the MTEX documentation (https://mtex-toolbox.github.io/CrystalOperations.html). In the lines below you can see my shortened script (find it attached also), focusing in the sections I am most interested on:

-----------------------------------------------------------------------------------------------------------------
```
%% No rotation
cs = crystalSymmetry('6/mmm'); % Mg-hex crystal symmetry
%%
m1 = Miller(1,0,-1,0,cs); % My crystallographic direction
% no symmetry
figure();
plot(m1,'labeled','backgroundColor','w','grid');
% symmetrised
s = unique(symmetrise(m1),'noSymmetry'); % the 6 unique symmetrically equivalent directions
figure();
plot(s,'grid','labeled','backgroundColor','w');
%% Rotated
ori = orientation.byEuler([10,20,30]*degree,cs); % Input orientation in Bunge Euler angles (deg)
%%
p = ori * s;
p1 = ori * m1;
figure();plot(p1,'labeled','grid')
figure();plot(p,'labeled','grid')
```
-----------------------------------------------------------------------------------------------------------------

Summing up the code, I am interested on identifying the points in the plot that correspond to each direction and its symmetrically equivalent directions. In the first part of the code I simply plot my crystallographic direction m1 in two cases: no symmetry and symmetrised. In the case of the symmetrised plot, the 'labeled' command allows to identify the point that corresponds to each direction with the subsequent Miller indices.

Then, in the last section of the code I want to apply this direction to a given orientation (defined by the Bunge Euler angles in degrees), in my case the orientation 'ori'. As well as in the first part of the code, I applied the rotation both with no symmetry and symmetrically. Then I plotted the result. My problem comes with the labeling of the points, which now is in the let's say 'vector3d' format. What I would want is to identify all the rotated parent directions with their corresponding Miller indices, as in the plot obtained in the first part.

I hope I have explained myself properly and someone can help me! Thanks in advance!!

Eugenia

**Comment by kilir** [comment, marked as answer]
Hi Eugenia,
please have a look here: https://groups.google.com/g/mtexmail/c/G5WSFmIwVd0/m/SpgXz9fhAwAJ
Does that answer your question?

Cheers,
Rüdiger

**Reply by eugenianietov** [reply]
Thank you Rüdiger, it was exactly what I was looking for!!

Cheers,
Eugenia
